cmd/packer-plugin-check/main.go

```diff
@@ -0,0 +1,178 @@
+// packer-plugin-check is a command used by plugins to validate compatibility and basic configuration
+// to work with Packer.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/packer/packer/plugin"
+)
+
+const packerPluginCheck = "packer-plugin-check"
+
+var (
+	hcl2spec = flag.Bool("hcl2spec", false, "flag to indicate that hcl2spec files should be checked.")
+	docs     = flag.Bool("docs", false, "flag to indicate that documentation files should be checked.")
+	load     = flag.String("load", "", "flag to check if plugin can be loaded by Packer.")
+)
+
+// Usage is a replacement usage function for the flags package.
+func Usage() {
+	fmt.Fprintf(os.Stderr, "Usage of "+packerPluginCheck+":\n")
+	fmt.Fprintf(os.Stderr, "\t"+packerPluginCheck+" [flags]\n")
+	fmt.Fprintf(os.Stderr, "Flags:\n")
+	flag.PrintDefaults()
+}
+
+func main() {
+	log.SetFlags(0)
+	log.SetPrefix(packerPluginCheck + ": ")
+	flag.Usage = Usage
+	flag.Parse()
+
+	if *hcl2spec == false && *docs == false && len(*load) == 0 {
```

Comment:
Since you are checking that at least one flag has been set you can check that `flag. NFlag()`  is 0 instead. 
```suggestion
	if flag. NFlag() == 0 {
```